### ISSUE TITLE
amazonka-elasticbeanstalk build failure

### ISSUE THREAD

**Issue by bergmark** [brendanhay/amazonka#251]
```
gen/Network/AWS/ElasticBeanstalk/Types/Sum.hs:57:24:
    Ambiguous occurrence ‘List’
    It could refer to either ‘Network.AWS.ElasticBeanstalk.Types.Sum.List’,
                             defined at gen/Network/AWS/ElasticBeanstalk/Types/Sum.hs:51:7
                          or ‘Network.AWS.Prelude.List’,
                             imported from ‘Network.AWS.Prelude’ at gen/Network/AWS/ElasticBeanstalk/Types/Sum.hs:20:1-36
                             (and originally defined in ‘lens-4.13:Control.Lens.Iso’)

gen/Network/AWS/ElasticBeanstalk/Types/Sum.hs:64:9:
    Ambiguous occurrence ‘List’
    It could refer to either ‘Network.AWS.ElasticBeanstalk.Types.Sum.List’,
                             defined at gen/Network/AWS/ElasticBeanstalk/Types/Sum.hs:51:7
                          or ‘Network.AWS.Prelude.List’,
                             imported from ‘Network.AWS.Prelude’ at gen/Network/AWS/ElasticBeanstalk/Types/Sum.hs:20:1-36
                             (and originally defined in ‘lens-4.13:Control.Lens.Iso’)
```

**Comment by bergmark** [comment]
Can also be seen on matrix: http://matrix.hackage.haskell.org/package/amazonka-elasticbeanstalk#GHC-7.10/amazonka-elasticbeanstalk-1.3.5

**Comment by brendanhay** [comment]
Thanks

**Comment by brendanhay** [comment]
I have a bit of an issue here, primarily fpco/stackage#975 and the related [stackage constraint](https://github.com/fpco/stackage/blob/9ce75d3554f24b4d57bca1f0415b3cd7eef00805/build-constraints.yaml#L1710). Since `retry` is pinned and `develop` of amazonka is dependent on `>= 0.7`, I assume everything is going to explode if I do a new release.

I can do a point release in the mean time to just get a build out, but it's less than ideal.

**Comment by bergmark** [comment]
It wouldn't explode :-) but we'd add an upper bound on `elasticbeanstalk` as well. It would be lifted along with `retry`'s.

So it doesn't matter that much for stackage at the moment.

**Comment by brendanhay** [comment]
Version `1.3.5.1` of `amazonka`, `amazonka-core`, and `amazonka-test` have been uploaded to Hackage. These contain the fix for the `amazonka-elasticbeanstalk` failure, but I haven't modified the prefered versions to force `amazonka-*` libraries to pick this version up (they all have `== 1.3.5.*` constraints) - I'm assuming this should be OK?

Also a restrictive upper bound for `retry` has been added. I'll need to upload `1.3.6` (which is already prepared) when `retry < 0.7` is removed from the stackage build constraints.

**Comment by bergmark** [comment]
Thanks!

I don't think stackage cares about preferred versions.

I will also temporarily ignore three test-suites that failed for the same reason
* amazonka-ec2
* amazonka-s3
* amasonka-swf

**Comment by brendanhay** [comment]
`amazonka-*-1.3.6` (now on Hackage) fixes the `lens-4.13` issues.
